src/DIPS.Xamarin.UI/Internal/Xaml/FloatingActionMenu.xaml.cs

```diff
@@ -0,0 +1,174 @@
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    [ContentProperty(nameof(Children))]
+    public partial class FloatingActionMenu : ContentView, IModalityHandler
+    {
+        private readonly double m_spaceBetween = 10;
+        private bool m_animationComplete = true;
+        private bool m_first = true;
+        private bool m_isExpanded;
+        private ModalityLayout? m_parent;
+        private double m_yTranslate;
+        internal readonly FloatingActionMenuBehaviour m_behaviour;
+
+        /// <summary>
+        ///     A floating action menu that can be added to either a RelativeLayout or an AbsoluteLayout. Add
+        ///     <see cref="MenuButton" />.
+        ///     as content.
+        /// </summary>
+        public FloatingActionMenu(FloatingActionMenuBehaviour behaviour)
+        {
+            m_behaviour = behaviour;
+            Children = new List<MenuButton>();
+            InitializeComponent();
+        }
+
+        private new List<MenuButton> Children { get; set; }
+
+        internal void ShowMenu(bool shouldShow)
+        {
+            if (m_animationComplete && m_isExpanded != shouldShow)
+            {
+                if (m_first)
+                {
+                    AdjustXPositions();
+                }
+
+                AnimateAll();
+            }
+        }
+
+        private void DisplayOverlay()
+        {
+            m_parent?.Show(this, Children.First());
+        }
+
+        private async Task AnimateAll()
+        {
+            m_animationComplete = false;
+
+            if (!m_isExpanded)
+            {
+                DisplayOverlay();
+            }
+            else
+            {
+                HideOverlay();
+            }
+
+            var multiplier = 1;
+            foreach (var menuButton in Children)
+            {
+                var maxOpacity = menuButton.IsEnabled ? 1 : .5;
+                menuButton.TranslateTo(0, m_isExpanded ? 0 : -m_yTranslate * multiplier, 250, Easing.CubicInOut);
+                menuButton.FadeTo(m_isExpanded ? 0 : maxOpacity, 250, Easing.CubicInOut);
+                multiplier += 1;
+            }
+
+            if (m_isExpanded)
+            {
+                ExpandButton.FadeTo(.5, 250, Easing.CubicInOut);
+            }
+            else
+            {
+                ExpandButton.FadeTo(1, 250, Easing.CubicInOut);
+            }
+            await ExpandButton.RelRotateTo(180, 250, Easing.CubicInOut);
+            m_isExpanded = !m_isExpanded;
+            m_animationComplete = true;
+        }
+
+        private void HideOverlay()
+        {
+            m_parent?.HideOverlay();
+        }
+
+        private async void ExpandButton_OnClicked(object sender, EventArgs e)
+        {
+            if (m_first)
+            {
+                AdjustXPositions();
+            }
+
+            if (m_animationComplete)
+            {
+                await AnimateAll();
+            }
+            m_behaviour.IsOpen = m_isExpanded;

```

Comment:
Isn't it better to just return when `m_animationComplete = true`?
Now the IsExpanded is swapped even though no animation would happend?